src/signals/my-incidents/components/EmailInput.tsx

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2022 Gemeente Amsterdam
+import { useRef, useState } from 'react'
+
+import { useHistory } from 'react-router-dom'
+import * as yup from 'yup'
+
+import Button from 'components/Button'
+import Label from 'components/Label'
+
+import { useMyIncidentContext } from '../context'
+import { routes } from '../definitions'
+import { usePostEmail } from '../hooks'
+import { ErrorWrapper, StyledErrorMessage, StyledInput } from './styled'
+import type { Error } from './types'
+
+const schema = yup.string().email().required().max(254)
+
+const validateInput = async (
+  inputRef: React.RefObject<HTMLInputElement>,
+  setValidation: (error: Error) => void
+) => {
+  const isValid = await schema
+    .isValid(inputRef?.current?.value)
+    .then((valid) => valid)
+
+  if (isValid) {
+    setValidation({
+      hasError: !isValid,
+      message: null,
+    })
+  } else {
+    setValidation({
+      hasError: !isValid,
+      message: 'Het veld moet een geldig e-mailadres bevatten',
+    })
+  }
+}
+
+export const EmailInput = () => {
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const [validation, setValidation] = useState<Error>({
+    hasError: false,
+    message: null,
+  })
+
+  const history = useHistory()
+  const [postEmail] = usePostEmail()
+
+  const { setEmail } = useMyIncidentContext()
+
+  const handleSubmit = () => {
+    validateInput(inputRef, setValidation)
```

Comment:
this is actually a good use case for react hook form. you just need to instantiate it, with your yup schema. Then on blur it will (configurable) submit your email, or after submit. This is a oneliner compared to the custom logic placed here. Its not the only way, but it would be more aligned with the rest of the application.

Reply:
The whole form logic can be as simple as 

```

const schema = yup.object({
  email: yup.email().required(),
}).required();

export default function SomeName() {
  const { register, handleSubmit, formState:{ errors } } = useForm({
    resolver: yupResolver(schema)
  });
  const onSubmit = data => console.log(data);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("email")} />
      <p>{errors.email?.message}</p>
      <input type="submit" />
    </form>
  );
}

```